lib/discovery_api/auth/auth0/secret_fetcher.ex

```diff
@@ -0,0 +1,57 @@
+defmodule DiscoveryApi.Auth.Auth0.SecretFetcher do
+  @moduledoc """
+  Fetches and caches the JWKS (JSON Web Key Set) from Auth0.
+  The JWKS is used to create the secret key needed to decode and validate JWTs (tokens from Auth0).
+  """
+  use Guardian.Token.Jwt.SecretFetcher
+
+  alias DiscoveryApi.Services.AuthService
+
+  def fetch_verifying_secret(_module, token_headers, _opts) do
+    %{"kid" => key_id} = token_headers
+
+    get_key(key_id)
+  end
+
+  defp get_key(key_id) do
+    case cached_jwks() |> key_from_jwks(key_id) do
+      {:error, _} -> fetch_and_cache_jwks() |> key_from_jwks(key_id)
+      result -> result
+    end
+  end
+
+  defp key_from_jwks(nil, _key_id) do
```

Comment:
Can we inline these bois? (This guy and the other one liner below)
```suggestion
  defp key_from_jwks(nil, _key_id), do: {:error, :jwks_unavailable}
```